packages/store/src/workspace/page.ts

```diff
@@ -813,9 +808,14 @@ export class Page extends Space<PageData> {
     if (model.flavour === 'affine:page') {
```

Comment:
We can change this to a type guard to avoid using `as` below:

```typescript
const isPageModel = (mode: BaseBlockModel): model is BlockSuiteInternal.BlockModels['affine:page'] => mode.flavour === 'affine:page'
```

Reply:
In fact I hava tried but  type guard not work as expected.

https://github.com/toeverything/blocksuite/blob/master/packages/global/src/utils.ts#L32

Reply:
The type declaration is not correct here, try something like:
```ts
type _K = keyof BlockSuiteInternal.BlockModels
type TupleToUnion<T extends unknown[]> = T[number];

export function matchFlavours<
  K extends _K,
  Key extends _K[] = _K[]
>(
  model: { flavour: K },
  expected: Key
): model is Pick<BlockSuiteInternal.BlockModels, TupleToUnion<Key>> & { flavour: K } {
  return expected.includes(model.flavour);
}
```

Reply:
It seems not work. 😢

![image](https://user-images.githubusercontent.com/50035259/221406720-fe1de3d4-ed0c-4ebb-afe8-6896410f189a.png)


![image](https://user-images.githubusercontent.com/50035259/221406709-fb1dcda9-218f-41c6-b31e-09c614e7cca3.png)

Reply:
Forget it, I'll try to figure it out.